lib/libc/Kconfig

```diff
@@ -32,4 +32,15 @@ config NEWLIB_LIBC_FLOAT_SCANF
 	  Build with floating point scanf enabled. This will increase the size of
 	  the image.
 
+config STDOUT_CONSOLE
+	bool
+	prompt "Send stdout to console"
+	depends on CONSOLE_HAS_DRIVER
+	default n
+	default y if NEWLIB_LIBC
```

Comment:
this should not set to default, even when newlib is used, printk might be the preferred way, leave this to the application.

Reply:
I agree with a should work comment, but that does not mean we enable it by default. For the should work we create tests, now all of the sudden everything built with newlib is using printf for example for logging, although it is not really needed. Applications should enable this explicitly. 

Think forward, we want to switch to newlib by default, this is being delayed, but that is the goal, so in this case we end up with printf enabled by default and we would get lots of "cant fit into memory" type of errors.

Reply:
> Think forward, we want to switch to newlib by default

For me, the situation is rather clear - we want to switch to newlib by default, because we want printf() and many other things work out of the box. Because if we don't want them to work out of the box, we should just stick to the toy minimal lib. But then if a user does enable a no-nonsense C library, all baseline stuff in that library should work out of the box, and what can be more baseline than printf()? If we don't do that, we just get another loop of "config hell", which Zephyr already has a bunch.


However, it's clear that this change needs to be split off to a separate PR, let me do that, and let's continue.

Reply:
> now all of the sudden everything built with newlib is using printf for example for logging

printf() should not be used for logging, definitely not of sudden. libc's stdio and Zephyr's logging should be kept separate, at least by default. Maybe, there should be an option, to log to printf(), and maybe an option to printf() to zephyr syslog, but default for both should be off.

Reply:
printf is actually not being used as a backend of syslog, it is set to printk. agree this should be its own PR with more input from other developers.